lib/Sai.cpp

```diff
@@ -228,19 +228,31 @@ sai_status_t Sai::set(
         // and do set on all if objectId == NULL
 
         bool success = true;
+        bool executed = false;
 
         for (auto& kvp: m_contextMap)
         {
+            if (objectType == SAI_OBJECT_TYPE_SWITCH && objectId != SAI_NULL_OBJECT_ID &&
+                    (attr->id == SAI_REDIS_SWITCH_ATTR_FLEX_COUNTER ||
+                     attr->id == SAI_REDIS_SWITCH_ATTR_FLEX_COUNTER_GROUP))
+            {
+                if (!kvp.second->m_redisSai->containsSwitch(objectId))
+                {
+                    continue;
+                }
+            }
+
             sai_status_t status = kvp.second->m_redisSai->set(objectType, objectId, attr);
 
             success &= (status == SAI_STATUS_SUCCESS);
+            executed = true;
```

Comment:
this flag is redundant, your condition on line 243 is exactly the same as in 236, please remove this

Reply:
Done